osu.Framework/Input/Handlers/MidiInputHandler.cs

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Commons.Music.Midi;
+using osu.Framework.Input.StateChanges;
+using osu.Framework.Logging;
+using osu.Framework.Platform;
+using osu.Framework.Threading;
+
+namespace osu.Framework.Input.Handlers
+{
+    public class MidiInputHandler : InputHandler
+    {
+        public override bool IsActive => true;
+        public override int Priority => 0;
+
+        private ScheduledDelegate scheduledRefreshDevices;
+
+        private readonly Dictionary<string, IMidiInput> openedDevices = new Dictionary<string, IMidiInput>();
+
+        public override bool Initialize(GameHost host)
+        {
+            // Try to initialize. This can throw on Linux if asound cannot be found.
+            try {
```

Comment:
Keep in mind we have open block in a new line
```suggestion
            try
            {

Reply:
I know, this PR is not yet done, I'll fix all AppVeyor and CodeFactor issues when it is. It's currently open for feedback related to code structure and implementation details, and visibility.

Reply:
> I know, this PR is not yet done, I'll fix all AppVeyor and CodeFactor issues when it is. It's currently open for feedback related to code structure and implementation details, and visibility.

Otherwise, this change is fantastic, great job on this one 👍